assets/js/googlesitekit/modules/datastore/modules.js

```diff
@@ -154,14 +174,92 @@ const baseActions = {
 
 		return { response, error };
 	},
+
+	/**
+	 * Registers a module
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string}          slug                         Module slug.
+	 * @param {Object}          settings                     Module settings.
+	 * @param {string}          [settings.name]              Optional. Module name. Default is the slug.
+	 * @param {string}          [settings.description]       Optional. Module description. Default empty string.
+	 * @param {string}          [settings.icon]              Optional. Module icon. Default empty string.
+	 * @param {number}          [settings.order]             Optional. Numeric indicator for module order. Default 10.
+	 * @param {string}          [settings.homepage]          Optional. Module homepage URL. Default empty string.
+	 * @param {WPElement}       [settings.settingsComponent] React component to render the settings panel. Default is the DefaultModuleSettings component.
+	 * @return {Object}                                      Redux-style action.
+	 */
+	*registerModule( slug, { settingsComponent = DefaultModuleSettings, ...settings } = {} ) {
+		invariant( slug, 'module slug is required' );
+
+		const registry = yield commonActions.getRegistry();
+		let registryKey = yield registry.select( CORE_SITE ).getRegistryKey();
+		if ( registryKey === undefined ) {
+			registryKey = uuidv4();
+			yield registry.dispatch( CORE_SITE ).setRegistryKey( registryKey );
+		}
```

Comment:
Shouldn't this block better be internal to the `core/site` store? I think it would simplify the consuming code if `getRegistryKey` handled the not set case on its own. If it's not set, create it itself so that the selector always returns one.